### PR TITLE
Fix: Throws if development server is used with build that does not contain tests

### LINKED ISSUE #124: Cannot read property 'policy' of undefined
This happens because of this new addition
`https://github.com/rwjblue/ember-cli-content-security-policy/blame/master/index.js#L133`

### PR DESCRIPTION
**Pull request by jelhan**
Fixes #124

### CODE REVIEW

**Comment by sandstrom** [comment]
Would an alternative solution be that people disabled this addon only for their test-environment? 

```js
module.exports = function(environment) {
  if (environment === 'test') {
    contentSecurityPolicy.enabled = false;
    // …

**Comment by jelhan** [comment]
> Would an alternative solution be that people disabled this addon only for their test-environment?

That's possible but won't solve the issue. Actually we want to support having the addon enabled for tests and still be able to serve builds not including tests:

- The addon must be enabled for test environment in order to get CSP validation checks as part of your test suite.
- `ember serve -prod` is a common scenario in which a build without tests is served by built-in web server.

File: index.js [hunk 1]
```diff
@@ -154,17 +160,25 @@ module.exports = {
...
+
+    if (policyObjectForTest) {
+      this._policyStringForTest = buildPolicyString(policyObjectForTest);
+    }
 
     app.use((req, res, next) => {
-      let isRequestForTests = req.originalUrl.startsWith('/tests');
+      // Use policy for test environment if
+      // - the request is for tests and
+      // - the build include tests
```
Comment (sandstrom) [inline index.js]: Minor, but could clarify:

```
      // Use policy for test environment if both of these conditions are met:
      // 1. the request is for tests and
      // 2. the build include tests
```

File: node-tests/e2e/test-support-test.js [hunk 2]
```diff
@@ -185,4 +185,28 @@ describe('e2e: provides test support', function() {
 
     expect(code).to.equal(0);
   });
+
+  it('does not break development server for builds not including tests', async function() {
```
Comment (sandstrom) [inline node-tests/e2e/test-support-test.js]:
```suggestion
 // One common scenario is when running the server in production mode locally, i.e. 
 // you are doing the production build on your local machine, connecting to your actual production server,
 // for example via `ember serve -prod`. In these cases we don't want this addon to break.
  it('does not break development server for builds not including tests', async function() {
```

File: index.js (hunk 1 above)
Comment (jelhan) [reply index.js]: Thanks a lot for your suggestion. Applied it in 952f3fccb4b9bf4b2d8a80cfd6b9991a25435eec.

File: node-tests/e2e/test-support-test.js (hunk 2 above)
Comment (jelhan) [reply node-tests/e2e/test-support-test.js]: Applied that suggestion on 952f3fccb4b9bf4b2d8a80cfd6b9991a25435eec as well.

**Pull request merged**